src/React.Core/Babel.cs

```diff
@@ -147,6 +193,23 @@ public virtual string TransformFile(string filename)
 			return output;
 		}
 
+        /// <summary>
+        /// Checks the in-memory cache. We need to invalidate any in-memory cache if there's no 
+        /// source map cached and forceGenerateSourceMap is true.
+        /// </summary>
+        /// <param name="cacheKey"></param>
+        /// <param name="forceGenerateSourceMap"></param>
+        /// <param name="cached"></param>
+        /// <returns>True if the cache contains the transformed source.</returns>
+        private bool CheckInMemoryCache(string cacheKey, bool forceGenerateSourceMap, out JavaScriptWithSourceMap cached)
+        {
+            cached = _cache.Get<JavaScriptWithSourceMap>(cacheKey);
```

Comment:
Instead of having an out parameter, what about just returning `null` if the cache isn't valid?

```
var cached = _cache.Get<JavaScriptWithSourceMap>(cacheKey);
return cached != null && (!forceGenerateSourceMap || cached.SourceMap != null
  ? cached
  : null;
```